### ISSUE TITLE
stomp: transaction abort broken after ack/nack

### ISSUE THREAD

**Issue by Anthchirp** [DiamondLightSource/python-workflows#3]
When a transaction is aborted after acking/nacking messages they can't be ack/nack'ed again, and are stuck in limbo. Only way to get rid of them is to restart the connection.

ActiveMQ bug. https://issues.apache.org/jira/browse/AMQ-6697

**Comment by Anthchirp** [comment]
Test script for validation
```python
from __future__ import division
import datetime
import logging
import stomp
import sys
import time

logger = logging.getLogger()
logger.setLevel(logging.WARN)
logger.addHandler(logging.StreamHandler())

class c(object):
  _stomp_listener = stomp.PrintingListener()

  def __init__(self):
    self.conn = stomp.Connection([( '127.0.0.1', 61613 )])
    self.conn.set_listener('', self._stomp_listener)
    self.conn.start()
    self.conn.connect('admin', 'password', wait=True)

    self._stomp_listener.on_message = self.receive
    errfun = self._stomp_listener.on_error

    def new_errfun(*args, **kwargs):
      errfun(*args, **kwargs)
      sys.exit(1)
    self._stomp_listener.on_error = new_errfun
    
    self.conn.send('/queue/some.queue', str(datetime.datetime.now()))

    self.conn.subscribe('/queue/some.queue', 1, ack='client-individual')

  def receive(self, header, message):
    print "received ", message

    self.conn.begin(transaction=1)
    self.conn.ack(header['message-id'], header['subscription'], transaction=1)
    self.conn.abort(1)
    self.conn.ack(header['message-id'], header['subscription'])

if __name__ == '__main__':
  c()

  time.sleep(3)
```

**Comment by Anthchirp** [comment]
same thing one level higher 
[workflows-test.txt](https://github.com/DiamondLightSource/python-workflows/files/1047695/workflows-test.txt)

**Comment by Anthchirp** [comment]
Fixed upstream for ActiveMQ 5.14.6 and 5.15
